### ISSUE TITLE
Async w/ ActiveJob silently fails

### ISSUE THREAD

**Issue by budnik** [getsentry/raven-ruby#506]
I just tried to setup `raven-ruby` following docs:
```ruby
Raven.configure do |config|
  config.async = ->(event) { SentryJob.perform_later(event.to_hash) }
end
```
It was silently failing to enqueue the job util I changed to:
```ruby
Raven.configure do |config|
  config.async = ->(event) { SentryJob.perform_later(event.to_hash.except(:request) }
end
```
I tried both `:inline` and `:sidekiq` AJ backends, I think there's some serialization issue that is swallowed silently. I'm going to provide repro gists soon meanwhile I opening this issue now to share this workaround.

**Comment by nateberkopec** [comment]
:( We should make sure we fail loudly if `async` is misconfigured.

**Comment by jeremy** [comment]
Hit this too. May be nice to fall back to non-async capture of both the event and the async exception.

**Comment by nateberkopec** [comment]
[FWIW we're getting `Thread.report_on_exception` in Ruby 2.4](https://bugs.ruby-lang.org/issues/6647).

**Comment by nateberkopec** [comment]
I think this is related to #511.

**Comment by nateberkopec** [comment]
PR for sync fallback is available at #548
